### ISSUE TITLE
control: support transport layer security for intra-AS connectivity

### ISSUE THREAD

**Issue by marcfrei** [scionproto/scion#4662]
To guarantee privacy, integrity, and authenticity of the DRKey-related RPCs, TLS should be enabled for the intra-AS control-plane APIs of the control service.

**Comment by oncilla** [comment]
Instead of integrating this functionally in the control service, you could also expose the control service functionality behind a reverse proxy (e.g. caddy) that can deal with AuthN/AuthZ. Then you also get TLS certificate management for free

**Comment by marcfrei** [comment]
Re TLS certificate management we also have to decide whether we want to rely on the Web-PKI or whether we would leverage the TRCs for this purpose too, right?

**Comment by JordiSubira** [comment]
Yes, indeed. We should decide on what PKI to use, something which is AS specific could work as well, relying that there's some type of Domain Controller and some sort of certificate distribution per host.

**Comment by katyatitkova** [comment]
Should we move it to discussions (or make a formal proposal out of it)?

**Comment by JordiSubira** [comment]
We can move it to discussions, if this is the way to go from now on for proposal-like topics.

**Comment by oncilla** [comment]
Just a thought:
I think this is also an aspect where we would profit from having something like the endhost API (that we talked about here https://scionproto.slack.com/archives/C8ADA9CEP/p1750164495411299)

Essentially, this API could easily be protected with TLS, and if desired, client authentication can be enforced on that level.
